src/Serialization/Fields.jl

```diff
@@ -48,3 +48,132 @@ function load_internal(s::DeserializerState, ::Type{gfp_fmpz_elem}, dict::Dict)
+        :parent => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{nf_elem, fq_nmod, Hecke.NfRelElem}},
+                       dict::Dict)
+    K = load_type_dispatch(s, dict[:parent], check_namespace=false)
+    polynomial = load_type_dispatch(s, dict[:polynomial], check_namespace=false)
+    return K(polynomial)
+end
+
+################################################################################
+# Non Simple Extension
+function save_internal(s::SerializerState, K::Union{NfAbsNS, NfRelNS})
+    def_pols = defining_polynomials(K)
+    return Dict(
+        :def_pols => save_type_dispatch(s, def_pols),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{NfAbsNS, NfRelNS}},
+                       dict::Dict)
+    def_pols = load_type_dispatch(s, dict[:def_pols], check_namespace=false)
+    K, _ = NumberField(def_pols, cached=false)
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState, k::Union{NfAbsNSElem, Hecke.NfRelNSElem})
+    K = parent(k)
+    polynomial = Oscar.Hecke.data(k)
+    polynomial_parent = parent(polynomial)
+    K_dict = save_type_dispatch(s, K)
+    return Dict(
+        :parent_field => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial),
+        :polynomial_parent => save_type_dispatch(s, polynomial_parent)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{NfAbsNSElem, Hecke.NfRelNSElem}},
+                       dict::Dict)
+    K = load_type_dispatch(s, dict[:parent_field]; check_namespace=false)
+    polynomial = load_type_dispatch(s, dict[:polynomial]; check_namespace=false)
+    polynomial = evaluate(polynomial, gens(K))
+
+    return K(polynomial)
+end
+
+
+################################################################################
+# FracField
+function save_internal(s::SerializerState, K::FracField)
+    return Dict(
+        :base_ring => save_type_dispatch(s, base_ring(K)),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: FracField},
+                       dict::Dict)
+    R, _ = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
+
+    return FractionField(R, cached=false)
+end
+
+# elements
+function save_internal(s::SerializerState, f::FracElem)
+    parent_dict = save_type_dispatch(s, parent(f))
+    return Dict(
+        :parent => parent_dict,
+        :den => save_type_dispatch(s, denominator(f)),
+        :num => save_type_dispatch(s, numerator(f))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: FracElem},
+                       dict::Dict)
+    R = load_type_dispatch(s, dict[:parent], check_namespace=false)
+    num = load_type_dispatch(s, dict[:num], check_namespace=false)
+    den = load_type_dispatch(s, dict[:den], check_namespace=false)
```

Comment:
Perhaps like this?
```suggestion
    num = load_type_dispatch(s, elem_type(R), dict[:num])
    den = load_type_dispatch(s, elem_type(R), dict[:den])
```

Reply:
this won't work since the denominator and numerator wont lie in R, this is why we cast them to R at the end 
```
julia> typeof(denominator(1 //x))
fmpq_poly

julia> typeof(parent(elem))
AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}}

julia> elem_type(parent(elem))
AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Frac{fmpq_poly}}
```

Reply:
this wont work due to this 
```
julia> elem_type(parent(1 //x)) == typeof(denominator(1 // x))
false
```
I guess i could do somethng like casting the denominator and numerator before saving?